foundation/plugins/org.jboss.tools.foundation.checkup/src/org/jboss/tools/foundation/checkup/JVMProblemDetector.java

```diff
@@ -517,4 +547,99 @@ public int hashCode() {
 			return toString().hashCode();
 		}
 	}
-}
+
+	private long eclipseStartTime = getEclipseStartTime();
+	private static final DateFormat DATE_FORMATTER = new SimpleDateFormat(LogEntry.F_DATE_FORMAT);
+	private Date currentDate = null;
+	
+	/**
+	 * for test purpose
+	 * @param date string in yyyy-MM-dd HH:mm:ss.SSS 
+	 * @throws ParseException
+	 */
+	public void setEclipseStartTime(String str) throws ParseException{
+		eclipseStartTime = DATE_FORMATTER.parse(str).getTime();
+	}
+	
+	private long getEclipseStartTime() {
```

Comment:
if you made this method protected you could just subclass it in tests to get the time you want and would not need to add a test only setEclipseStartTime method.